src/libmongoc/src/mongoc/mongoc-structured-log-command-private.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mongoc-prelude.h"
+#include "mongoc-structured-log.h"
+#include "mongoc-cmd-private.h"
+
+#ifndef MONGOC_STRUCTRURED_LOG_COMMAND_PRIVATE_H
+#define MONGOC_STRUCTRURED_LOG_COMMAND_PRIVATE_H
+
+typedef struct {
+   const char *command_name;
+   const char *db_name;
+   const bson_t *command;
+   const bson_t *reply;
+   const bson_error_t *error;
+   int64_t duration;
+   int64_t operation_id;
+   uint32_t request_id;
+   uint32_t driver_connection_id;
+   uint32_t server_connection_id;
+   bool explicit_session;
+} _mongoc_structured_log_command_t;
+
+void
+mongoc_structured_log_command_started (const bson_t *command,
+                                       const char *command_name,
+                                       const char *db_name,
+                                       int64_t operation_id,
+                                       uint32_t request_id,
+                                       uint32_t driver_connection_id,
+                                       uint32_t server_connection_id,
+                                       bool explicit_session);
```

Comment:
I believe this might not be capturing OP_MSG document sequences.
There are two ways of initializing a `mongoc_apm_command_started_t`:
- `mongoc_apm_command_started_init` - initialized from a `bson_t` command
- `mongoc_apm_command_started_init_with_cmd` - initialized from a `mongoc_cmd_t`, and appends the OP_MSG document sequence (see note in command monitoring spec: https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst#implementation-notes)

I'm not sure off-hand if it's feasible to change this to accept a `mongoc_cmd_t` to handle commands with and without OP_MSG document sequence. If not, we may need a separate `mongoc_structured_log_command_started_with_cmd`.

Reply:
I originally had the method accept only a `mongoc_cmd_t`, but had to change it to accommodate legacy commands that are run without a `mongoc_cmd_t`. I'll add a separate `mongoc_structured_log_command_started_with_cmd`, similar to how we have two initialisers for `mongoc_apm_command_started_t`.